indexing-service/src/test/java/org/apache/druid/indexing/common/task/CompactionTaskParallelRunTest.java

```diff
@@ -219,6 +219,37 @@ public void testRunParallelWithRangePartitioning()
     }
   }
 
+  @Test
+  public void testRunParallelWithRangePartitioningWithSingleTask()
+  {
+    // Range partitioning is not supported with segment lock yet
+    if (lockGranularity == LockGranularity.SEGMENT) {
+      return;
+    }
```

Comment:
[JUnit has an `Assume` API](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html), which could be a good fit here.